<a href="https://colab.research.google.com/github/pavlihinav/Theoriaofprobability_HW/blob/main/%D0%94%D0%976%D0%A1%D1%80%D0%B0%D0%B2%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5_%D0%B4%D0%BE%D0%BB%D0%B5%D0%B9_%D0%9F%D0%BE%D1%81%D1%82%D1%80%D0%BE%D0%B5%D0%BD%D0%B8%D0%B5_%D0%B4%D0%BE%D0%B2%D0%B5%D1%80%D0%B8%D1%82%D0%B5%D0%BB%D1%8C%D0%BD%D0%BE%D0%B3%D0%BE_%D0%B8%D0%BD%D1%82%D0%B5%D1%80%D0%B2%D0%B0%D0%BB%D0%B0.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Задача 1.**

 Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.

Найти доверительный интервал для оценки математического ожидания с надежностью
0.95,
если выборочная средняя M = 80, а объем выборки n = 256.

In [1]:
import numpy as np
from statsmodels.stats.weightstats import _tconfint_generic as t_stat
from scipy import stats

In [2]:
m = 80
sigma = 16
n = 256
p = 0.95
alpha = 0.05
z = stats.norm.ppf(alpha/2)

In [3]:
(m - z * sigma / np.sqrt(n), m + z * sigma / np.sqrt(n))

(81.95996398454005, 78.04003601545995)

Доверительный интервал: [81.96; 78.04].

**Задача 2.**


 В результате 10 независимых измерений некоторой величины X, выполненных с
одинаковой точностью,получены опытные данные:

6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1

Предполагая, что результаты измерений подчинены нормальному закону распределения
вероятностей,оценить истинное значение величины X при помощи доверительного интервала,покрывающего это значение с доверительной вероятностью 0,95.


In [21]:
x = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
x_mean = x.mean()
x_std = x.std(ddof = 1)
x_mean_std = x_std / (np.sqrt(len(x)))

result = t_stat(x_mean, x_mean_std, len(x) - 1, 1 - 0.95, 'two-sided')

In [24]:
print(f"Доверительный интервал величины Х: {result}")

Доверительный интервал величины Х: (6.267515851415713, 6.912484148584288)


**Задача 3.**

 Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170

Рост матерей 178, 165, 165, 173, 168, 155, 160, 164, 178, 175


Используя эти данные построить 95% доверительный интервал для разности среднего
роста родителей и детей.

In [26]:
mothers = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])
daughters = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])

difference_of_means = np.mean(mothers) - np.mean(daughters)
standard_error = np.sqrt(np.var(mothers, ddof=1)/len(mothers) + np.var(daughters, ddof=1)/len(daughters))
t_critical = stats.t.ppf(0.975, len(mothers) + len(daughters) - 2)
l_border = difference_of_means - t_critical * standard_error
r_border = difference_of_means + t_critical * standard_error

print(f'>>> 95% доверительный интервал для разности среднего роста родителей и детей: [{l_border:.2f}, {r_border:.2f}]')

>>> 95% доверительный интервал для разности среднего роста родителей и детей: [-6.27, 10.07]
